packages/mdc-textfield/foundation.js

```diff
@@ -278,6 +310,51 @@ class MDCTextFieldFoundation extends MDCFoundation {
     }
   }
 
+  /**
+   * Styles the Text Field component's root Element based on the optional
+   * params. If a param is not supplied, the method makes no changes related to
+   * that param.
+   * @param {boolean=} optIsValid
+   * @param {boolean=} optIsFocused
+   * @param {boolean=} optIsDisabled
+   * @private
+   */
+  styleRoot_(optIsValid, optIsFocused, optIsDisabled) {
```

Comment:
This can be broken up:
* handleValidityChange(optIsValid)
* handleFocusChange(optIsFocused)
* handleEnabledChange(optIsDisabled)

They seem to be separable, and I always prefer to de-couple things.